admin-console/src/main/webapp/src/components/CreateEditModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { ActionGroup, Button, Form, FormGroup, HelperText, HelperTextItem, Modal, ModalVariant,
+    Spinner, TextArea, TextInput } from '@patternfly/react-core';
+import React from 'react';
+
+import { EventType } from '../types/Notifications';
+
+interface CreateEditModalProps {
+    isEdit: boolean;
+    showModal: boolean;
+    applicationName?: string;
+    initialEventType?: EventType;
+    isLoading?: boolean;
+    onClose: () => void;
+    onSubmit: () => void;
+    eventTypeQuery: unknown;
+
+}
+
+export const CreateEditModal: React.FunctionComponent<CreateEditModalProps> = (props) => {
+    const [ eventType, setEventType ] = React.useState<Partial<EventType>>(props.initialEventType ?? {});
```

Comment:
This is good.
Now we need to pass it to the `onSubmit` by doing something like:
```
const onSubmitLocal = React.useCallback(() => { props.onSubmit(eventType)  }, [ eventType, props.onSubmit]) // might require to make this more verbose to make the linter happy
```

Reply:
Something else that we might want to do is to initialize this value everytime the isOpen or maybe the initialEventType changes:
```
React.useEffect(() => {
  setEventType(props.initialEventType);
}, [ props.initialEventType ]);
```